File: Codes/2598-smallest-missing-non-negative-integer-after-operations.go
/*
 * @Author: LetMeFly
 * @Date: 2025-10-16 19:40:17
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-16 20:01:33
 */
package main

func findSmallestInteger(nums []int, value int) (ans int) {
    cnt := map[int]int{}
    for _, t := range nums {
        cnt[(t % value + value) % value]++
    }
    for cnt[ans % value] > 0 {
        cnt[ans % value]--
        ans++
    }
    return
}

File: Codes/2598-smallest-missing-non-negative-integer-after-operations.java
/*
 * @Author: LetMeFly
 * @Date: 2025-10-16 19:40:17
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-10-16 19:59:10
 */
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        Map<Integer, Integer> cnt = new HashMap<>();
        for (int t : nums) {
            cnt.merge((t % value + value) % value, 1, Integer::sum);
        }
        int ans = 0;
        while (cnt.merge(ans % value, -1, Integer::sum) >= 0) {  // 记得是大于等于
            ans++;
        }
        return ans;
    }
}